### PR TITLE
feat: Sphinx docs build action

### PR DESCRIPTION
**Pull request by ivadym**
### This PR:

Github actions that builds the documentation using Sphinx.

Address the failing of readthedoc #773 

**Context**:
- Fixed versions from #807.
- Sphinx action: https://github.com/ammaraskar/sphinx-action 

### Review and tests: 
- [ ] Tests pass
- [ ] Code review
- [ ] New code is executed and covered by tests, and test approve

### CODE REVIEW

**Comment by ivadym** [comment]
[Successful build](https://github.com/Clinical-Genomics/BALSAMIC/runs/4106270983?check_suite_focus=true) after fixing docs requirements ([failing previously](https://github.com/Clinical-Genomics/BALSAMIC/runs/4106195925?check_suite_focus=true)).

**Comment by ivadym** [comment]
[Successful build](https://github.com/Clinical-Genomics/BALSAMIC/runs/4106270983?check_suite_focus=true) after fixing docs requirements ([failing previously](https://github.com/Clinical-Genomics/BALSAMIC/runs/4106195925?check_suite_focus=true)).

File: CHANGELOG.rst [hunk 1]
```diff
@@ -12,6 +12,7 @@ Added:
 * Delly is added as a submodule and removed from rest of the conda environments #787
 * Added `.,PASS` to all structural variant filter rules to resolve the issues with missing calls in filtered file
 * Handling of QC metrics validation errors #783
+* Github Action workflow that builds the docs using Sphinx #809
```
Comment (ashwini06) [inline CHANGELOG.rst:15]: Can you also update the following in 
 Fixed: 
"* ReadtheDocs building failure due to dependencies, fixed by locking versions #773*

**Review by ashwini06 — CHANGES_REQUESTED** [review]
Let's see what readthedocs displays after merging 👍🏼

**Review by ashwini06 — APPROVED** [review]

**Pull request merged**